salt/gis.sls

```diff
@@ -0,0 +1,127 @@
+# get the software packages installed
+base:
+  pkg.installed:
+    - install_recommends: false
+    - pkgs:
+      # just some basic packages, not specific to the carto setup.
+      - git 
+      - unzip
+      - curl
+      - build-essential
+      - software-properties-common
+    - refresh: true
+
+# installing the required GIS-software, plus...
+# create the postgresql database and enabling PostGIS
+gis:
+  pkg.installed:
+    - install_recommends: false
+    - pkgs:
+      # here come the goodies; PostGIS is elementary to a current dev. environment.
+      - postgresql-9.3-postgis-2.1
+      - postgresql-contrib-9.3
+      - proj-bin
+      - libgeos-dev
+  # Let's make it easy for our GIS-in-aid software to access our PostGIS database
+  file.blockreplace:
+    - name: /etc/postgresql/9.3/main/pg_hba.conf
+    - marker_start: "# TYPE  DATABASE"
+    - marker_end: '# "local" is'
+    - content: "local   all             gisuser                                 trust"
+  service:
+    - name: postgresql
+    - running
+    - watch:
+      - file: postgresql.conf.work_mem
+      - file: postgresql.conf.maintenance_work_mem
+      - file: gis
+    - require:
+      - pkg: gis
+      - file: gis
+  postgres_user.present:
+    - name: gisuser
+    - superuser: true
+    - createdb: true
+    - require:
+      - service: gis
+  postgres_database.present:
+    - name: gis
+    - db_user: gisuser
+    - require:
+      - postgres_user: gis
+
+# PostGIS: hstore -- extension
+hstore:
+  postgres_extension.present:
+    - db_user: gisuser
+    - maintenance_db: gis
+    - require:
+        - service: gis
+
+# PostGIS: postgis -- extension
+postgis:
+  postgres_extension.present:
+    - db_user: gisuser
+    - maintenance_db: gis
+    - require:
+      - service: gis
+
+# Conservative tweaks to PostgreSQL conf, based on 
+# "Loading OSM data" located at switch2osm.org.
+# TODO: figure out if it's possible to get rid of the other ID, so I can sleep better at night.
```

Comment:
Other ID?

fwiw, I've started changing postgres settings by adding `include_dir = 'conf.d'` to postgresql.conf and then dumping my custom settings in conf.d/base.conf. Probably not an issue here, but it means there's only one change in postgresql.conf if I later upgrade it.